billing/models.py

```diff
@@ -1,3 +1,91 @@
+import datetime
+from typing import Any
+
+from django.core.exceptions import ValidationError
 from django.db import models
+from django.utils import timezone
+
+from billing.common.constants import BillingConstants
+from billing.common.constants import CurrencyConstants
+from billing.common.strings import AccountingMethodText
+from billing.common.strings import BillingDetailsAbstractModelText
+from billing.common.strings import BillingValidationErrorText
+from managers.models import Project
+from users.common.fields import ChoiceEnum
+from users.models import CustomUser
+from utils.models import BaseModel
+
+
+class BillingDetailsAbstractQuerySet(models.QuerySet):
+    pass
+
+
+class BillingDetailsAbstract(BaseModel):
+    class Meta:
+        abstract = True
+
+    class Currency(ChoiceEnum):
+        PLN = CurrencyConstants.CurrencyDict["PLN"]["Code"]
+        USD = CurrencyConstants.CurrencyDict["USD"]["Code"]
+        EUR = CurrencyConstants.CurrencyDict["EUR"]["Code"]
+
+    class AccountingMethod(ChoiceEnum):
+        HOURLY = AccountingMethodText.HOURLY
+        MONTHLY = AccountingMethodText.MONTHLY
+
+    user = models.ForeignKey(CustomUser, on_delete=models.PROTECT, related_name="%(class)s")
+    value = models.PositiveSmallIntegerField(BillingDetailsAbstractModelText.VALUE, blank=False, null=False)
+    currency = models.CharField(
+        max_length=BillingConstants.CURRENCY_MAX_LENGTH.value, choices=Currency.choices(), default=Currency.PLN
+    )
+    accounting_method = models.CharField(
+        max_length=BillingConstants.ACCOUNTING_METHOD_MAX_LENGTH.value,
+        choices=AccountingMethod.choices(),
+        default=AccountingMethod.HOURLY.name,
+    )
+    from_date = models.DateTimeField(
+        BillingDetailsAbstractModelText.FROM_DATE, auto_now_add=False, blank=False, null=False
```

Comment:
Do we need to add `auto_now_add=False`? Shouldn't it be `False` by default?

Reply:
That's true. Removed